lib/msf/core/opt.rb

```diff
@@ -30,17 +30,21 @@ def self.CPORT(default=nil, required=false, desc="The local client port")
     def self.LHOST(default=nil, required=true, desc="The listen address")
       Msf::OptAddressLocal.new(__method__.to_s, [ required, desc, default ])
     end
-	
-	# @return [OptAddress]
-    def self.DOMAIN(default=nil, required=true, desc="The listen address")
-      Msf::OptAddress.new(__method__.to_s, [ required, desc, default ])
+    
+    # @return [OptAddress]
+    def self.DOMAIN(default=nil, required=true, desc="Domain name")
+      Msf::OptString.new(__method__.to_s, [ required, desc, default ])
```

Comment:
Clever hackery, not sure if this is the best way to implement since every module load would instantiate these opts for every domain name. Also not sure about using a name such as "default" for anything. Still though, clever :)